rust/datafusion/src/execution/context.rs

```diff
@@ -100,12 +102,30 @@ impl ExecutionContext {
     }
 
     /// Register a table so that it can be queried from SQL
-    pub fn register_table(&mut self, name: &str, provider: Rc<Table>) {
+    pub fn register_table(&mut self, name: &str, provider: Rc<TableProvider>) {
         self.datasources
             .borrow_mut()
             .insert(name.to_string(), provider);
     }
 
+    /// Get a table by name
+    pub fn table(&mut self, table_name: &str) -> Result<Arc<Table>> {
+        match self.datasources.borrow().get(table_name) {
+            Some(provider) => {
+                Ok(Arc::new(TableImpl::new(Arc::new(LogicalPlan::TableScan {
+                    schema_name: "".to_string(),
```

Comment:
It seems odd to always use an empty string, should `schema_name` be an `Option<String>`